### PR TITLE
atf: Support loading mmc FIP on imx8mm

### PR DESCRIPTION
**Pull request by niej**
Update rev to support loading mmc FIP image on imx8mm, so that fip
can be flashed to SD directly and updated separately.

Signed-off-by: Jun Nie <jun.nie@linaro.org>

### CODE REVIEW

File: meta-freescale-mbl/conf/machine/imx8mmevk-mbl.conf [hunk 1]
```diff
@@ -118,9 +118,11 @@ MACHINE_FIRMWARE_remove = " linux-firmware-ath10k"
 ##############################################################################
 MBL_WKS_BOOTLOADER1_FILENAME = "imx-boot-imx8mmevk-mbl-sd.bin-flash_evk"
 MBL_WKS_BOOTLOADER1_OFFSET_BANK1_KiB = "33"
-MBL_WKS_BOOTLOADER1_SIZE_KiB = "1024"
+MBL_WKS_BOOTLOADER1_SIZE_KiB = "991"
 
-MBL_WKS_BOOTLOADER2_SKIP = "1"
+MBL_WKS_BOOTLOADER2_FILENAME = "fip.bin"
+MBL_WKS_BOOTLOADER2_SIZE_KiB = "2048"
```
Comment (diego-sueiro) [inline meta-freescale-mbl/conf/machine/imx8mmevk-mbl.conf]: I can see that for other platforms the size is set to `1024`.
Checking the size of the u-boot binary for imx8 it is around 655K.
Can we set `1024` here?

Comment (jonathanhaigh-arm) [reply meta-freescale-mbl/conf/machine/imx8mmevk-mbl.conf]: ...Or maybe I should have set it to a higher value elsewhere.

Comment (diego-sueiro) [reply meta-freescale-mbl/conf/machine/imx8mmevk-mbl.conf]: So, what is the decision? Increase to `2048` for other platforms, decrease this one to `1024` or leave as is?

Comment (niej) [reply meta-freescale-mbl/conf/machine/imx8mmevk-mbl.conf]: fip.bin is 1065937, including optee and u-boot. So 1024K is too small for sure.

Comment (jonathanhaigh-arm) [reply meta-freescale-mbl/conf/machine/imx8mmevk-mbl.conf]: We should also leave room for future changes... How about we say 4096KiB on all MACHINEs?

Comment (jonathanhaigh-arm) [reply meta-freescale-mbl/conf/machine/imx8mmevk-mbl.conf]: Or even 16MiB?

Comment (diego-sueiro) [reply meta-freescale-mbl/conf/machine/imx8mmevk-mbl.conf]: @jonathanhaigh-arm, no problem. Would you like to raise a ticket to address this?

Comment (jonathanhaigh-arm) [reply meta-freescale-mbl/conf/machine/imx8mmevk-mbl.conf]: I'll use https://jira.arm.com/browse/IOTMBL-696.

**Comment by bryanodonoghue** [comment]
@niej - MMC changes break imx6ul but, discussed with @diego-sueiro we will merge anyway and keep the two platforms isolated with different SHAs

**Comment by niej** [comment]
http://jenkins.mbed-linux.arm.com/view/JunNie/job/jun-imx8mm-atf-load-mmc-fip/2/

lava test is still in progress...

**Pull request merged**